packages/@aws-cdk/aws-codestarnotifications/lib/rule.ts

```diff
@@ -0,0 +1,262 @@
+   *
+   * You must specify this property (either via props or via `addTarget`)
+   *
+   * @default - No targets.
+   */
+  readonly targets?: IRuleTarget[];
+
+  /**
+   * A list of event types associated with this notification rule.
+   * For a complete list of event types and IDs, see Notification concepts in the Developer Tools Console User Guide.
+   * @see https://docs.aws.amazon.com/dtconsole/latest/userguide/concepts.html#concepts-api
+   *
+   * You must specify this property (either via props or via `addEvents`)
+   *
+   * @default - No events.
+   */
+  readonly events?: string[];
+}
+
+/**
+ * Represents a notification rule
+ */
+export interface IRule extends IResource {
+
+  /**
+   * The ARN of the notification rule (i.e. arn:aws:codestar-notifications:::notificationrule/01234abcde)
+   * @attribute
+   */
+  readonly ruleArn: string;
+
+  /**
+   * The name of the notification rule
+   * @attribute
+   */
+  readonly ruleName: string;
+}
+
+/**
+ * A new notification rule
+ *
+ * @resource AWS::CodeStarNotifications::NotificationRule
+ */
+export class Rule extends Resource implements IRule {
+  /**
+   * Import an existing notification rule provided an ARN
+   * @param scope The parent creating construct
+   * @param id The construct's name
+   * @param notificationRuleArn Notification rule ARN (i.e. arn:aws:codestar-notifications:::notificationrule/01234abcde)
+   */
+  public static fromNotificationRuleArn(scope: Construct, id: string, notificationRuleArn: string): IRule {
+    const parts = Stack.of(scope).parseArn(notificationRuleArn);
+    class Import extends Resource implements IRule {
+      readonly ruleArn = notificationRuleArn;
+      readonly ruleName = parts.resourceName || '';
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * @attribute
+   */
+  readonly ruleArn: string;
+
+  /**
+   * @attribute
+   */
+  readonly ruleName: string;
+
+  /**
+   * The source config of notification rule
+   */
+  readonly source: RuleSourceConfig;
+
+  /**
+   * The target config of notification rule
+   */
+  readonly targets: RuleTargetConfig[] = [];
+
+  /**
+   * The events of notification rule
+   */
+  readonly events: string[] = [];
+
+  constructor(scope: Construct, id: string, props: RuleProps) {
+    super(scope, id);
+
+    this.source = this.bindSource(props.source);
+
+    if (props.events) {
+      this.addEvents(props.events);
+    }
+
+    props.targets?.forEach((target) => {
+      this.addTarget(target);
+    });
+
+    this.ruleName = props.ruleName || this.generateName();
+
+    this.ruleArn = new CfnNotificationRule(this, 'Resource', {
+      name: this.ruleName,
+      status: props?.status,
+      detailType: props.detailType || DetailType.FULL,
+      targets: this.targets,
+      eventTypeIds: this.events,
+      resource: this.source.sourceAddress,
+    }).ref;
```

Comment:
Assign the `CfnNotificationRule` to a local variable instead:

```suggestion
    const resource = new CfnNotificationRule(this, 'Resource', {
      name: this.ruleName,
      status: props?.status,
      detailType: props.detailType || DetailType.FULL,
      targets: this.targets,
      eventTypeIds: this.events,
      resource: this.source.sourceAddress,
    });
    this.ruleArn = resource.ref;

Reply:
Already adjusted